engine/src/main/java/org/hibernate/validator/internal/metadata/PredefinedScopeBeanMetaDataManager.java

```diff
@@ -30,13 +30,17 @@
 import org.hibernate.validator.internal.util.ExecutableHelper;
 import org.hibernate.validator.internal.util.ExecutableParameterNameProvider;
 import org.hibernate.validator.internal.util.classhierarchy.ClassHierarchyHelper;
+import org.hibernate.validator.internal.util.classhierarchy.Filters;
+import org.hibernate.validator.metadata.BeanMetaDataClassNormalizer;
 
 public class PredefinedScopeBeanMetaDataManager implements BeanMetaDataManager {
 
+	private final BeanMetaDataClassNormalizer beanMetaDataClassNormalizer;
+
 	/**
 	 * Used to cache the constraint meta data for validated entities
 	 */
-	private final Map<String, BeanMetaData<?>> beanMetaDataMap;
+	private final Map<Class<?>, BeanMetaData<?>> beanMetaDataMap;
```

Comment:
My memory is blurry, but didn't we key this intentionally _not_ by class objects to avoid any CL leaks caused by a container-managed VF kept after an application got undeployed?

Reply:
The standard BeanMetaDataManager is keyed by class. I had issues with the predefined scope one so I used strings for a while but it's now fixed so we can use classes too now.